script.js

```diff
@@ -1,4 +1,135 @@
+  var randomInteger = Math.floor(randomDecimal);
+
+  // it's a number from 0 - 2, add 1. can be 1,2,3
+  var choiceNumber = randomInteger + 1;
+
+  return choiceNumber;
+};
+
+// This takes computer choice random number and decodes it into scissors, paper, or stone
+var choiceDecoder = function (numberGenerator) {
+  if (numberGenerator == 3) {
+    return "scissors";
+  } else if (numberGenerator == 2) {
+    return "paper";
+  } else if (numberGenerator == 1) {
+    return "stone";
+  } else {
+    return "Invalid input";
+  }
+};
+
+// This takes user choice input and encodes it into 1, 2, or 3
+var choiceEncoder = function (input) {
+  if (input == "scissors" || input == "reversed scissors") {
+    return 3;
+  } else if (input == "paper" || input == "reversed paper") {
+    return 2;
+  } else if (input == "stone" || input == "reversed stone") {
+    return 1;
+  } else {
+    return "Invalid input";
+  }
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  var myOutputValue = "";
+
+  // Waiting for username input
+  if (gameState == "Pre-username mode") {
+    userName = input;
+    if (userName == "") {
+      myOutputValue = "Hello! What is your name?";
+      return myOutputValue;
+    } else {
+      gameState = "Username mode";
+    }
+  }
+
+  // After username input, first prompt
+  if (gameState == "Username mode") {
+    userName = input;
+    gameState = "Game mode";
+    myOutputValue = `Hey welcome ${userName}! Make a choice! Scissors, paper, or stone?`;
+    return myOutputValue;
+  } else if (gameState == "Game mode") {
+    // Game mode
+    var userChoiceNumber = choiceEncoder(input);
+    var comChoiceNumber = numberGenerator();
+    var comChoice = choiceDecoder(comChoiceNumber);
+
+    var outcome = userChoiceNumber - comChoiceNumber;
+
+    myOutputValue = `You chose ${input}. <br> Computer chose ${comChoice}. <br>`;
+
+    if (isNaN(outcome)) {
+      // Check if input is valid
+      return `What was that?! <br><br> ${userName}, please input a valid choice: "scissors", "paper", or "stone"?`;
+    } else if (input == "scissors" || input == "paper" || input == "stone") {
+      totalTries += 1;
+      // Logic flow below - normal game
+
+      if (outcome == 0) {
+        totalDraws += 1;
+        myOutputValue += `It's a draw!`;
+      } else if (outcome == 1 || outcome == -2) {
+        totalWins += 1;
+        myOutputValue += `You won!`;
+      } else if (outcome == 2 || outcome == -1) {
+        totalLosses += 1;
+        myOutputValue += `You lost!`;
+      }
+    } else {
+      // Reversed game flow
+      totalTries += 1;
+      if (outcome == 0) {
+        totalDraws += 1;
+        myOutputValue += `It's a draw!`;
+      } else if (outcome == 1 || outcome == -2) {
+        totalLosses += 1;
+        myOutputValue += `You lost!`;
+      } else if (outcome == 2 || outcome == -1) {
+        totalWins += 1;
+        myOutputValue += `You won!`;
+      }
+    }
```

Comment:
To challenge yourself, you could also abstract away this entire `outcome` portion of the code into a function that immediately sets the global score variables and determines the outcome message.